P0009/P0009.md

```diff
@@ -889,6 +891,52 @@ might only be known at run time.
 This would make a rank-3 array with two compile-time extents
 a natural data structure for the matrices.
 
+Different extents will result in different `mdspan` instantiations,
+even if the element type, rank, layout, and accessor are all the same.
+Some users may be concerned that this will cause a proliferation of types.
+Users with this concern can restrict themselves to only dynamic extents
+and the default layout and accessor.
+They can also "type erase" all three layouts we propose here
+to a single layout type, `layout_stride`.
+
+Our view is that forcing too much type erasure
+(e.g., only permitting dynamic extents,
+or making layout or accessor run-time polymorphic) would hinder performance.
+C++ has generally been moving towards providing more information at compile time.
+Our experience is that compilers optimize array access better
+when given more information about strides at compile time.
+
+We made `mdspan`'s element type, extents, layout, and accessor orthogonal
+in order to simplify customization.
+Even if changing `mdspan`'s design to mix these policies together
+would reduce the number of instantiations,
+we would not do it, because it would make customization harder.
```

Comment:
this doesn't make much sense. How does mixing reduce instantiations if you don't reduce the amount of information encoded?

Reply:
Right, exactly : - )  I'm not saying it makes sense, rather that even if it were somehow possible, it wouldn't be what we want.